UM/View/GL/OpenGL.py

```diff
@@ -52,25 +102,32 @@ def getGPUType(self):
     #   is used to make direct OpenGL calls so should match OpenGL as closely
     #   as possible.
     def getBindingsObject(self):
-        raise NotImplementedError("Should be implemented by subclasses")
+        return self._gl
 
     ##  Create a FrameBuffer Object.
     #
     #   This should return an implementation-specifc FrameBufferObject subclass.
     def createFrameBufferObject(self, width, height):
-        raise NotImplementedError("Should be implemented by subclasses")
+        return FrameBufferObject.FrameBufferObject(width, height)
 
     ##  Create a Texture Object.
     #
     #   This should return an implementation-specifc Texture subclass.
     def createTexture(self):
-        raise NotImplementedError("Should be implemented by subclasses")
+        return Texture.Texture(self._gl)
 
     ##  Create a ShaderProgram Object.
     #
     #   This should return an implementation-specifc ShaderProgram subclass.
     def createShaderProgram(self, file_name):
-        raise NotImplementedError("Should be implemented by subclasses")
+        shader = ShaderProgram.ShaderProgram()
+        # The version_string must match the keys in shader files.
+        if OpenGLContext.isLegacyOpenGL():
+            version_string = ""
```

Comment:
Actually, I like it how it is currently. "fragment" and "vertex" are the baseline, and "fragment41core", "vertex41core", "geometry41core" are the special ones. I've added a comment to make it clearer.